debugger/server/debugger_server.go

```diff
@@ -79,17 +79,48 @@ func (ds *DebugServer) windowResizeHandler() error {
 	}
 }
 
+func (ds *DebugServer) getIP() string {
+	if runtime.GOOS == "darwin" {
+		// macOS doesn't have a docker0 bridge
+		return "127.0.0.1"
+	}
+
+	iface, err := net.InterfaceByName("docker0")
+	if err != nil {
+		ds.console.Warnf("falling back to 0.0.0.0 due to docker0 lookup error: %v", err.Error())
+		return "0.0.0.0"
+	}
+	addrs, err := iface.Addrs()
+	if err != nil {
+		ds.console.Warnf("falling back to 0.0.0.0 due to docker0 addrs error: %v", err.Error())
+		return "0.0.0.0"
+	}
+	for _, a := range addrs {
+		switch v := a.(type) {
+		case *net.IPNet:
+			if x := v.IP.To4(); x != nil {
+				return x.String()
+			}
+		}
+	}
+
+	ds.console.Warnf("falling back to 0.0.0.0 due to docker0 addrs being empty")
+	return "0.0.0.0"
+}
+
 // Start starts the debug server listener
 func (ds *DebugServer) Start() (string, error) {
-	l, err := net.Listen("tcp", ds.addr)
+	addr := fmt.Sprintf("%s:0", ds.getIP())
+
+	l, err := net.Listen("tcp", addr)
 	if err != nil {
-		return "", errors.Wrap(err, fmt.Sprintf("failed listening on %s", ds.addr))
+		return "", errors.Wrap(err, fmt.Sprintf("failed listening on %s", addr))
 	}
 
 	go ds.windowResizeHandler()
 
 	go func() {
-		ds.console.Printf("interactive debugger listening\n")
+		ds.console.Printf("interactive debugger listening on %v\n", l.Addr())
```

Comment:
Nit (for consistency)

```suggestion
		ds.console.Printf("Interactive debugger listening on %v\n", l.Addr())
```

Thinking that we should also add a prefix for this one, same way we use a prefix for `buildkitd` messages. What do you think? eg `ds.console.WithPrefix('debugger').Printf('....')`